common/configuration/manager.go

```diff
@@ -75,6 +76,10 @@ func New[R Role](ctx context.Context, router Router[R], providers []Provider[R])
 	return m, nil
 }
 
+func (m *Manager[R]) ProviderKeyForAccessor(accessor *url.URL) string {
```

Comment:
This can probably be a static function, but it is also fine.

Reply:
Already ran into a need for this to be static in another branch. Switching it over now.